### PR TITLE
Namespace should be optional for `state shell`.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-1091" title="DX-1091" target="_blank">StoryDX-1091</a>  `state shell` should work on current dir and global default
  </td></table>
  <br />
 

If no namespace is given, start a shell in the current project or the default project.

### CODE REVIEW

File: internal/runners/shell/shell.go
Comment (Naatan) [inline internal/runners/shell/shell.go:80]: I think you can replace most/all of this with: https://github.com/ActiveState/cli/blob/a2e3ebf27dfe163a69e7172fc4770ef14b71c5bb/pkg/projectfile/projectfile.go#L774

Comment (mitchell-as) [reply internal/runners/shell/shell.go]: Awesome, thanks. You saved me some time.

File: test/integration/shell_int_test.go
Comment (MDrakos) [inline test/integration/shell_int_test.go:77]: nit: There could be a few more tests that cover the cases in the ACs. This is testing that it sources the global default but we could also have tests for the current project, namespace, current project when there is currently a global default and potentially more.

**Review by MDrakos — COMMENTED** [review]
Looks good. Just leaving a comment for my nit and I'm not sure if this will auto merge when I approve.

File: test/integration/shell_int_test.go
Comment (mitchell-as) [reply test/integration/shell_int_test.go:77]: This behavior hinges on `projectfile.GetProjectFilePath()`, which I'm almost positive is tested as a side-effect in other integration tests like `state activate`. I didn't think the additional explicit tests are worth ballooning the integration test runtime even more. What do you think @Naatan ?

**Comment by mitchell-as** [comment]
Tagging David for review since Mike is out.

File: internal/runners/shell/shell.go
Comment (Naatan) [inline internal/runners/shell/shell.go:64]:
```suggestion
	if params.Namespace.Project != "" {
```

The owner on its own doesn't seem like it would work.

File: internal/runners/shell/shell.go
Comment (Naatan) [inline internal/runners/shell/shell.go:70]: This is too generic of an error. Plus we weren't attempting to do anything with the shell yet.

File: internal/runners/shell/shell.go
Comment (Naatan) [inline internal/runners/shell/shell.go:77]:
```suggestion
		proj, err = project.GetOnce()
```

File: test/integration/shell_int_test.go
Comment (Naatan) [reply test/integration/shell_int_test.go:79]: I agree this test is a little bare. It'd be nice if we can test the "project from CWD" and "project from namespace" variations. 

While technically this is a little redundant with activate, there is enough plumbing here that they aren't identical.

It would also be good if we can test that it actually is in a subshell. The current test just "waits for input" which doesn't assert whether it is now in a subshell.

**Review by Naatan — CHANGES_REQUESTED** [review]

File: test/integration/shell_int_test.go
Comment (mitchell-as) [reply test/integration/shell_int_test.go:80]: Okay, I added a cwd test. The `cp.Expect("Activated")` is the assertion that we're in a subshell, and the `cp.SendLine("exit")` and subsequent `cp.ExpectExitCode(0)` indirectly verify we've exited the subshell. (If we were not in a subshell, those lines would fail.)

The "project from namespace" variation tests already exist.

**Review by daved — APPROVED** [review]

**Review by daved — APPROVED** [review]

**Comment by mitchell-as** [comment]
@Naatan ping. I cannot merge this without your approval overriding a previous change request.

**Review by Naatan — APPROVED** [review]

**Pull request merged**
